### DISCUSSION TITLE
如何让一个Class里的属性只对某个xaml可访问，或类似此功能的做法？

### DISCUSSION THREAD

**Discussion by Lj-dog** [BYJRK/DotNet-Discussions#368] [WPF 开发]
最近遇到一个与此视频误区一相关的问题。[【MVVM常见误区之：滥用自动属性、可点击的控件、单选项的绑定】 https://www.bilibili.com/video/BV1Q94y1e7qV/?share_source=copy_web&vd_source=12c510f4de2a2c910d478336fceb8ff5](url)

  **情景**：我在为**一个model写datatemplate**的时候，发现有很多经过运算的临时变量需要绑定到控件里或Geometry里的Point来画图。比如外部new我这model时，会给出两个点，然后我计算出中点或其他点后通过datatemplate显示。

  **问题**：
一、这些临时变量如果是写在后台cs绑定到前台xaml是需要public或internal的，那这时我的model里就会有**一堆外部用不到的pubulic属性**，感觉像视频里误区一一样很蠢，而且别人看到总会问起。
二、要是写Convert把临时变量的运算放到前台xaml里，工程可能比一显得更臃肿，而且有些**变量需要重复使用**。
三、我也看过一期视频是将一些文件放到另外的类库dll里，然后写internal的。但这两个文件单独放个dll是不是有点不合适，而且还有别人一起开发。

  **想法**：
  目前是想在xaml里写个变量来绑定一个表达式，然后在datatemplate里重复使用。
例如：
```
//cs  
Class Model{  
public int A;  
public int B;
public Model(int a,int b)
{
  A=a;B=b;
}
}
```
```
//xaml 
<sys:Int x:Key="sum" > <Binding A+B> </sys:Int>
<TextBlock Text={Binding  sum}>
``` 
 A，B为model里面的pubulic属性。 然后这sum会根据A，B的值变化来运行A+B然后变化。

不知道有没有这类功能呢？

**Comment by CodingOctocat** [comment]
用这个：[levitali/CompiledBindings](https://github.com/levitali/CompiledBindings) 或者 [CalcBinding](https://github.com/Alex141/CalcBinding)。

要么就把临时属性封装成一个类作为 Model 的属性，加上注释就行了。

**Reply by Lj-dog** [reply]
CalcBinding完美解决. 
你说的第二个方法是: 将用到的临时变量,封装成一个类结构吗? 如 Class Model {   Class Temp{} } , 但还是能调用到把

**Reply by CodingOctocat** [reply]
封装成一个至少看着舒心了，加个注释让别人不要访问或者添加 `[Obsolete("不要访问", error: true)]` 特性，面向约定编程🤣

**Comment by WHYBBE** [comment]
有时候用用自动属性挺好的，放过自己，放过MVVM😎

**Reply by Lj-dog** [reply]
会有一堆public的属性,这样对合作开发者不友好.
